File: Allegro5 Test/BlackFloor.cpp
#include "BlackFloor.h"


BlackFloor::BlackFloor(void)
{
}


void BlackFloor::Init(ALLEGRO_BITMAP *image = 0, float x = 0, float y = 0, int boundX = 100, int boundY = 100){
	Floor::Init(image, x, y, boundX, boundY);
	SetCollideType(BLACK);
}

BlackFloor::~BlackFloor(void)
{
}

File: js/blackFloor.js
// BlackFloor class - floor that only black character can stand on

class BlackFloor extends Floor {
    constructor() {
        super();
        this.setCollideType(COLLIDE_TYPE.BLACK);
    }
    
    init(texture, x, y, boundX, boundY) {
        super.init(texture, x, y, boundX, boundY);
        
        // Make the sprite appear black/dark
        if (this.sprite) {
            this.sprite.tint = 0x333333; // Dark gray/black tint
        }
    }
}